ui/src/shared/components/TimeSeries.tsx

```diff
@@ -355,7 +361,9 @@ const mstp = (state: AppState, props: OwnProps) => {
   ]
 
   return {
+    appState: state,
```

Comment:
hrrmmm. does this mean this page is rerendering each time any part of appstate changes?

Reply:
That's not something I had considered. I'm afraid I don't know enough about the mechanics of React to know the answer to this.

Reply:
since `getRunQueryResults` is only interested in using state to `getAllVariables(state)`, maybe we can just consume what is returned from `getAllVariables(state)`... ? or..... perhaps `getRunQueryResults` should be a thunk?

Reply:
🤔 that may work. I think I had shied away from making it a thunk originally since it was causing problems with API request + pending results from that API request being stored in redux, but now that we're not storing it in a reducer that might be a possibility? Let me run it by @drdelambre to see what he thinks

Reply:
👍

Reply:
This worked! Just pushed up the changes. I tested to see if the cache broke in unexpected ways (it didn't), and I also did a few screen grabs for performance differences (there weren't). Great call!